src/audio/mux/mux.c

```diff
@@ -610,48 +610,116 @@ static int mux_copy(struct comp_dev *dev)
 
 static int mux_reset(struct comp_dev *dev)
 {
+	int dir = dev->pipeline->source_comp->direction;
```

Comment:
Silly question here. So MUX is a component that allows to select one output stream from multiple input sources?

Reply:
Mux defined in "ASoC sense" will do what you said. Sof mux can basically pick any channels from all input streams (also duplication), _but_ it will not mix any input channels. So it does a little bit more, than you might see in some ASoC slides in the internet. Hence it is not defined as asoc mux, but as generic processing component.